diff -ruN atlas-3.10.1/CONFIG/src/atlconf_misc.c atlas-3.10-new/CONFIG/src/atlconf_misc.c
--- atlas-3.10.1/CONFIG/src/atlconf_misc.c	2013-01-08 11:15:42.000000000 -0700
+++ atlas-3.10-new/CONFIG/src/atlconf_misc.c	2014-03-03 15:40:18.749258400 -0700
@@ -710,6 +710,18 @@
 	 p[1] = ' ';
 	 p += 2;
       }
+      else if (path[i] == '(')
+      {
+         *p = '\\';
+         p[1] = '(';
+         p += 2;
+      }
+      else if (path[i] == ')')
+      {
+         *p = '\\';
+         p[1] = ')';
+         p += 2;
+      }
       else
          *p++ = path[i];
    }
diff -ruN atlas-3.10.1/CONFIG/src/Makefile atlas-3.10-new/CONFIG/src/Makefile
--- atlas-3.10.1/CONFIG/src/Makefile	2013-01-08 11:15:42.000000000 -0700
+++ atlas-3.10-new/CONFIG/src/Makefile	2014-03-03 15:40:18.749258400 -0700
@@ -434,31 +434,31 @@
 DoNothing:
 
 make_subdirs:
-	mkdir src bin tune interfaces
-	cd src ; mkdir testing auxil blas lapack pthreads threads
+	mkdir -p src bin tune interfaces
+	cd src ; mkdir -p testing auxil blas lapack pthreads threads
 	cd src/blas ; \
-           mkdir f77reference reference gemv ger gemm kbmm \
+           mkdir -p f77reference reference gemv ger gemm kbmm \
                  level1 level2 level3 pklevel3
-	cd src/blas/reference ; mkdir level1 level2 level3
-	cd src/blas/level2 ; mkdir kernel
-	cd src/blas/pklevel3 ; mkdir gpmm sprk
-	cd src/blas/level3 ; mkdir rblas kernel
-	cd src/pthreads ; mkdir blas misc
-	cd src/pthreads/blas ; mkdir level1 level2 level3
-	cd src/threads ; mkdir blas lapack
-	cd src/threads/blas ; mkdir level3 level2
-	cd tune ; mkdir blas sysinfo lapack threads
-	cd tune/blas ; mkdir gemm gemv ger level1 level3
-	cd interfaces ; mkdir blas lapack
-	cd interfaces/lapack ; mkdir C F77
-	cd interfaces/lapack/C ; mkdir src testing
-	cd interfaces/lapack/F77 ; mkdir src testing
-	cd interfaces/blas ; mkdir C F77
-	cd interfaces/blas/C ; mkdir src testing
-	cd interfaces/blas/F77 ; mkdir src testing
-	cd interfaces/lapack ; mkdir C2F
-	cd interfaces/lapack/C2F ; mkdir src
-	mkdir ARCHS
+	cd src/blas/reference ; mkdir -p level1 level2 level3
+	cd src/blas/level2 ; mkdir -p kernel
+	cd src/blas/pklevel3 ; mkdir -p gpmm sprk
+	cd src/blas/level3 ; mkdir -p rblas kernel
+	cd src/pthreads ; mkdir -p blas misc
+	cd src/pthreads/blas ; mkdir -p level1 level2 level3
+	cd src/threads ; mkdir -p blas lapack
+	cd src/threads/blas ; mkdir -p level3 level2
+	cd tune ; mkdir -p blas sysinfo lapack threads
+	cd tune/blas ; mkdir -p gemm gemv ger level1 level3
+	cd interfaces ; mkdir -p blas lapack
+	cd interfaces/lapack ; mkdir -p C F77
+	cd interfaces/lapack/C ; mkdir -p src testing
+	cd interfaces/lapack/F77 ; mkdir -p src testing
+	cd interfaces/blas ; mkdir -p C F77
+	cd interfaces/blas/C ; mkdir -p src testing
+	cd interfaces/blas/F77 ; mkdir -p src testing
+	cd interfaces/lapack ; mkdir -p C2F
+	cd interfaces/lapack/C2F ; mkdir -p src
+	mkdir -p ARCHS
 
 #
 # Simple test returns error if directory doesn't exist
diff -ruN atlas-3.10.1/CONFIG/src/probe_comp.c atlas-3.10-new/CONFIG/src/probe_comp.c
--- atlas-3.10.1/CONFIG/src/probe_comp.c	2013-01-08 11:15:42.000000000 -0700
+++ atlas-3.10-new/CONFIG/src/probe_comp.c	2014-03-03 15:37:29.037561300 -0700
@@ -1293,6 +1293,7 @@
       ln = CheckStrLen(ln, &lnlen, i);
       sprintf(ln, "find %s -maxdepth 1 -name '*gcc*' -exec ./xisgcc '{}' \\;",
               sp);
+      fprintf(stderr, "Executing: %s", ln);
       free(sp);
       fp = atlsys(NULL, ln, verb, 1);
       if (fp)
@@ -1391,6 +1392,7 @@
       ln = CheckStrLen(ln, &lnlen, i);
       sprintf(ln,
          "find %s -maxdepth 1 -name '*gfortran*' -exec ./xisgcc '{}' \\;", sp);
+      printf("Executing: %s", ln);
       gccs = GetLinesFromFile(atlsys(NULL, ln, verb, 1), gccs);
       i = SelectBestGcc(verb, targ, cmaj, cmin, gccs);
       if (i > 1)
@@ -1448,8 +1450,8 @@
    {
       i = 64 + strlen(sp);
       ln = CheckStrLen(ln, &lnlen, i);
-      sprintf(ln,
-         "find %s -maxdepth 1 -name '*gfortran*' -exec ./xisgcc '{}' \\;", sp);
+      sprintf(ln, "find %s -maxdepth 1 -name '*gfortran*' -exec ./xisgcc '{}' \\;", sp);
+      fprintf(stderr, "Executing: %s", ln);
       free(sp);
       gccs = GetLinesFromFile(atlsys(NULL, ln, verb, 1), gccs);
       i = SelectBestGcc(verb, targ, cmaj, cmin, gccs);
@@ -1549,6 +1551,7 @@
       i = 64 + lcmp + strlen(sp);
       ln = CheckStrLen(ln, &lnlen, i);
       sprintf(ln, "find %s -maxdepth 1 -name '%s'", sp, comp);
+      fprintf(stderr, "Executing: %s", ln);
       free(sp);
       gccs = GetLinesFromFile(atlsys(NULL, ln, verb, 1), gccs);
       free(ln);
diff -ruN atlas-3.10.1/Make.top atlas-3.10-new/Make.top
--- atlas-3.10.1/Make.top	2013-01-08 11:15:40.000000000 -0700
+++ atlas-3.10-new/Make.top	2014-03-03 15:40:18.759258500 -0700
@@ -9,12 +9,12 @@
 	$(BZIP) error_$(ARCH).tar
 
 leafstart:
-	cd $(leaf) ; ln -s $(BLDdir)/Make.inc Make.inc
+	cd $(leaf) ; ln -sf $(BLDdir)/Make.inc Make.inc
 
 build:
 	cd bin/ ; $(MAKE) xatlas_build
 	cd bin/ ; ./xatlas_build $(INSTFLAGS)
-time: 
+time:
 	./xatlbench -dc $(BLDdir)/bin/INSTALL_LOG -dp $(BLDdir)/ARCHS/$(ARCH)
 C_sanity_test:
 	@ echo "BUILDING C TESTERS ONLY (GET F77 FOR MORE COMPLETE TESTING):"
@@ -89,7 +89,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_PT_PT \
                  $(BINdir)/LAPACK_TEST/SUMMARY_pt_pt.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_PT_PT
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_PT_PT
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(PTLAPACKlib)" \
            BLASLIB="$(PTF77BLASlib) $(PTCBLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_PT_PT/$(ARCH).out 2>&1
@@ -104,7 +104,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_PT_AB \
                  $(BINdir)/LAPACK_TEST/SUMMARY_pt_ab.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_PT_AB
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_PT_AB
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(PTLAPACKlib)" \
            BLASLIB="$(F77BLASlib) $(CBLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_PT_AB/$(ARCH).out 2>&1
@@ -119,7 +119,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_PT_SB \
                  $(BINdir)/LAPACK_TEST/SUMMARY_pt_sb.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_PT_SB
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_PT_SB
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(PTLAPACKlib)" \
            BLASLIB="$(BLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_PT_SB/$(ARCH).out 2>&1
@@ -134,7 +134,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_PT_FB \
                  $(BINdir)/LAPACK_TEST/SUMMARY_pt_fb.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_PT_FB
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_PT_FB
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(PTLAPACKlib)" \
            BLASLIB="$(FBLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_PT_FB/$(ARCH).out 2>&1
@@ -149,7 +149,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_AL_PT \
                  $(BINdir)/LAPACK_TEST/SUMMARY_al_pt.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_AL_PT
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_AL_PT
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(LAPACKlib)" \
            BLASLIB="$(PTF77BLASlib) $(PTCBLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_AL_PT/$(ARCH).out 2>&1
@@ -164,7 +164,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_AL_AB \
                  $(BINdir)/LAPACK_TEST/SUMMARY_al_ab.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_AL_AB
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_AL_AB
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(LAPACKlib)" \
            BLASLIB="$(F77BLASlib) $(CBLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_AL_AB/$(ARCH).out 2>&1
@@ -179,7 +179,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_AL_SB \
                  $(BINdir)/LAPACK_TEST/SUMMARY_al_sb.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_AL_SB
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_AL_SB
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(LAPACKlib)" \
            BLASLIB="$(BLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_AL_SB/$(ARCH).out 2>&1
@@ -194,7 +194,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_AL_FB \
                  $(BINdir)/LAPACK_TEST/SUMMARY_al_fb.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_AL_FB
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_AL_FB
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(LAPACKlib)" \
            BLASLIB="$(FBLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_AL_FB/$(ARCH).out 2>&1
@@ -209,7 +209,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_SL_PT \
                  $(BINdir)/LAPACK_TEST/SUMMARY_sl_pt.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_SL_PT
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_SL_PT
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(SLAPACKlib)" \
            BLASLIB="$(PTF77BLASlib) $(PTCBLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_SL_PT/$(ARCH).out 2>&1
@@ -224,7 +224,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_SL_AB \
                  $(BINdir)/LAPACK_TEST/SUMMARY_sl_ab.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_SL_AB
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_SL_AB
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(SLAPACKlib)" \
            BLASLIB="$(F77BLASlib) $(CBLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_SL_AB/$(ARCH).out 2>&1
@@ -239,7 +239,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_SL_SB \
                  $(BINdir)/LAPACK_TEST/SUMMARY_sl_sb.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_SL_SB
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_SL_SB
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(SLAPACKlib)" \
            BLASLIB="$(BLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_SL_SB/$(ARCH).out 2>&1
@@ -254,7 +254,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_SL_FB \
                  $(BINdir)/LAPACK_TEST/SUMMARY_sl_fb.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_SL_FB
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_SL_FB
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(SLAPACKlib)" \
            BLASLIB="$(FBLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_SL_FB/$(ARCH).out 2>&1
@@ -269,7 +269,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_FL_PT \
                  $(BINdir)/LAPACK_TEST/SUMMARY_fl_pt.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_FL_PT
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_FL_PT
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(FLAPACKlib)" \
            BLASLIB="$(PTF77BLASlib) $(PTCBLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_FL_PT/$(ARCH).out 2>&1
@@ -284,7 +284,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_FL_AB \
                  $(BINdir)/LAPACK_TEST/SUMMARY_fl_ab.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_FL_AB
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_FL_AB
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(FLAPACKlib)" \
            BLASLIB="$(F77BLASlib) $(CBLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_FL_AB/$(ARCH).out 2>&1
@@ -299,7 +299,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_FL_SB \
                  $(BINdir)/LAPACK_TEST/SUMMARY_fl_sb.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_FL_SB
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_FL_SB
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(FLAPACKlib)" \
            BLASLIB="$(BLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_FL_SB/$(ARCH).out 2>&1
@@ -314,7 +314,7 @@
 	cd $(BINdir)/LAPACK_TEST ; make cleanup
 	- rm -rf $(BINdir)/LAPACK_TEST/LAOUT_FL_FB \
                  $(BINdir)/LAPACK_TEST/SUMMARY_fl_fb.txt
-	mkdir $(BINdir)/LAPACK_TEST/LAOUT_FL_FB
+	mkdir -p $(BINdir)/LAPACK_TEST/LAOUT_FL_FB
 	cd $(BINdir)/LAPACK_TEST ; make ATL_all LAPACKLIB="$(FLAPACKlib)" \
            BLASLIB="$(FBLASlib) $(ATLASlib) $(LIBS)" \
            | tee LAOUT_FL_FB/$(ARCH).out 2>&1
@@ -325,7 +325,7 @@
 	if [ -d "$(BINdir)/AtlasTest/$(ARCH)" ]; then \
            rm -rf $(BINdir)/AtlasTest/$(ARCH)/* ; \
         else \
-           mkdir $(BINdir)/AtlasTest/$(ARCH) ; \
+           mkdir -p $(BINdir)/AtlasTest/$(ARCH) ; \
         fi
 	cd $(BINdir)/AtlasTest/$(ARCH) ; ../configure --atldir=$(BLDdir)
 scope_full_results: force_build
@@ -337,7 +337,7 @@
 	$(MAKE) -f Make.top $(BINdir)/AtlasTest/$(ARCH)
 	cd $(BINdir)/AtlasTest/$(ARCH) ; make | tee $(ARCH).out 2>&1
 	$(MAKE) -f Make.top scope_full_results
-           
+
 full_test_nh :$(BINdir)/AtlasTest
 	$(MAKE) -f Make.top $(BINdir)/AtlasTest/$(ARCH)
 	cd $(BINdir)/AtlasTest/$(ARCH) ; nohup make | tee $(ARCH).out 2>&1
@@ -351,12 +351,12 @@
 	echo $(L1Size) > $(SYSdir)/res/L1CacheSize
 
 results:
-	mkdir results
-	mkdir results/charts
-	mkdir results/tvecs
-	mkdir results/raw
+	mkdir -p results
+	mkdir -p results/charts
+	mkdir -p results/tvecs
+	mkdir -p results/raw
 EXtest:
-	mkdir EXtest
+	mkdir -p EXtest
 startup : lib include results EXtest
 	$(MAKE) -f Make.top leafstart leaf=interfaces/lapack/C2F/src
 	$(MAKE) -f Make.top leafstart leaf=interfaces/lapack/F77/src
@@ -404,36 +404,36 @@
 	$(MAKE) -f Make.top leafstart leaf=lib
 	$(MAKE) -f Make.top leafstart leaf=bin
 	cd src/threads ; touch atomic.inc
-	cd src/blas/gemm ; mkdir KERNEL
-	cd tune/blas/gemm ; mkdir KERNEL
-	cd tune/blas/gemm ; mkdir res
-	cd tune/blas/gemv ; mkdir res
-	cd tune/blas/ger ; mkdir res
-	cd tune/blas/level1 ; mkdir res
-	cd tune/blas/level1 ; mkdir GEN
-	cd tune/blas/level3 ; mkdir res
-	cd tune/sysinfo ; mkdir res
-	cd tune/lapack  ; mkdir res
-	cd tune/threads ; mkdir res
-	cd bin ; mkdir INSTALL_LOG
-	cd tune/blas/gemv ; ln -s $(SRCdir)/tune/blas/gemv/MVTCASES MVTCASES
-	cd tune/blas/gemv ; ln -s $(SRCdir)/tune/blas/gemv/MVNCASES MVNCASES
-	cd tune/blas/ger ; ln -s $(SRCdir)/tune/blas/ger/R1CASES R1CASES
-	cd tune/blas/ger ; ln -s $(SRCdir)/tune/blas/ger/R2CASES R2CASES
-	cd tune/blas/gemv ; ln -s $(SRCdir)/tune/blas/gemv/CASES CASES
-	cd tune/blas/gemm ; ln -s $(SRCdir)/tune/blas/gemm/CASES CASES
-	cd tune/blas/level1 ; ln -s $(SRCdir)/tune/blas/level1/SWAP SWAP
-	cd tune/blas/level1 ; ln -s $(SRCdir)/tune/blas/level1/SET SET
-	cd tune/blas/level1 ; ln -s $(SRCdir)/tune/blas/level1/SCAL SCAL
-	cd tune/blas/level1 ; ln -s $(SRCdir)/tune/blas/level1/ROT ROT
-	cd tune/blas/level1 ; ln -s $(SRCdir)/tune/blas/level1/NRM2 NRM2
-	cd tune/blas/level1 ; ln -s $(SRCdir)/tune/blas/level1/IAMAX IAMAX
-	cd tune/blas/level1 ; ln -s $(SRCdir)/tune/blas/level1/DOT DOT
-	cd tune/blas/level1 ; ln -s $(SRCdir)/tune/blas/level1/CPSC CPSC
-	cd tune/blas/level1 ; ln -s $(SRCdir)/tune/blas/level1/COPY COPY
-	cd tune/blas/level1 ; ln -s $(SRCdir)/tune/blas/level1/AXPY AXPY
-	cd tune/blas/level1 ; ln -s $(SRCdir)/tune/blas/level1/AXPBY AXPBY
-	cd tune/blas/level1 ; ln -s $(SRCdir)/tune/blas/level1/ASUM ASUM
+	cd src/blas/gemm ; mkdir -p KERNEL
+	cd tune/blas/gemm ; mkdir -p KERNEL
+	cd tune/blas/gemm ; mkdir -p res
+	cd tune/blas/gemv ; mkdir -p res
+	cd tune/blas/ger ; mkdir -p res
+	cd tune/blas/level1 ; mkdir -p res
+	cd tune/blas/level1 ; mkdir -p GEN
+	cd tune/blas/level3 ; mkdir -p res
+	cd tune/sysinfo ; mkdir -p res
+	cd tune/lapack  ; mkdir -p res
+	cd tune/threads ; mkdir -p res
+	cd bin ; mkdir -p INSTALL_LOG
+	cd tune/blas/gemv ; ln -sf $(SRCdir)/tune/blas/gemv/MVTCASES MVTCASES
+	cd tune/blas/gemv ; ln -sf $(SRCdir)/tune/blas/gemv/MVNCASES MVNCASES
+	cd tune/blas/ger ; ln -sf $(SRCdir)/tune/blas/ger/R1CASES R1CASES
+	cd tune/blas/ger ; ln -sf $(SRCdir)/tune/blas/ger/R2CASES R2CASES
+	cd tune/blas/gemv ; ln -sf $(SRCdir)/tune/blas/gemv/CASES CASES
+	cd tune/blas/gemm ; ln -sf $(SRCdir)/tune/blas/gemm/CASES CASES
+	cd tune/blas/level1 ; ln -sf $(SRCdir)/tune/blas/level1/SWAP SWAP
+	cd tune/blas/level1 ; ln -sf $(SRCdir)/tune/blas/level1/SET SET
+	cd tune/blas/level1 ; ln -sf $(SRCdir)/tune/blas/level1/SCAL SCAL
+	cd tune/blas/level1 ; ln -sf $(SRCdir)/tune/blas/level1/ROT ROT
+	cd tune/blas/level1 ; ln -sf $(SRCdir)/tune/blas/level1/NRM2 NRM2
+	cd tune/blas/level1 ; ln -sf $(SRCdir)/tune/blas/level1/IAMAX IAMAX
+	cd tune/blas/level1 ; ln -sf $(SRCdir)/tune/blas/level1/DOT DOT
+	cd tune/blas/level1 ; ln -sf $(SRCdir)/tune/blas/level1/CPSC CPSC
+	cd tune/blas/level1 ; ln -sf $(SRCdir)/tune/blas/level1/COPY COPY
+	cd tune/blas/level1 ; ln -sf $(SRCdir)/tune/blas/level1/AXPY AXPY
+	cd tune/blas/level1 ; ln -sf $(SRCdir)/tune/blas/level1/AXPBY AXPBY
+	cd tune/blas/level1 ; ln -sf $(SRCdir)/tune/blas/level1/ASUM ASUM
 	$(MAKE) -f Make.top refresh arch=$(arch)
 
 refresh :
@@ -590,10 +590,10 @@
 	cd bin ; rm -f *.grd
 
 lib :
-	mkdir lib
+	mkdir -p lib
 
 include:
-	mkdir include
+	mkdir -p include
 	touch include/atlas_ssyr2NX.h
 	touch include/atlas_ssyrNX.h
 	touch include/atlas_dsyr2NX.h
@@ -659,11 +659,11 @@
 	cp $(LIBdir)/libcblas.a $(INSTdir)/.
 	cp $(LIBdir)/liblapack.a $(INSTdir)/.
 	chmod 0644 $(INSTdir)/libatlas.a  $(INSTdir)/liblapack.a \
-                   $(INSTdir)/libcblas.a 
+                   $(INSTdir)/libcblas.a
 	- cp $(LIBdir)/libf77blas.a $(INSTdir)/.
 	- chmod 0644 $(INSTdir)/libf77blas.a
 	- cp $(LIBdir)/libptcblas.a $(INSTdir)/.
-	- chmod 0644 $(INSTdir)/libptcblas.a 
+	- chmod 0644 $(INSTdir)/libptcblas.a
 	- cp $(LIBdir)/libptf77blas.a $(INSTdir)/.
 	- chmod 0644 $(INSTdir)/libptf77blas.a
 	- cp $(LIBdir)/libsatlas.dylib $(INSTdir)/.
