diff -ruN boost-1_50_0/boost/math/special_functions/sinc.hpp boost-1_50_0-new/boost/math/special_functions/sinc.hpp
--- boost-1_50_0/boost/math/special_functions/sinc.hpp	2009-01-07 08:37:59.000000000 -0800
+++ boost-1_50_0-new/boost/math/special_functions/sinc.hpp	2012-11-24 01:42:18.000000000 -0800
@@ -67,7 +67,7 @@
             static T const    taylor_2_bound = sqrt(taylor_0_bound);
             static T const    taylor_n_bound = sqrt(taylor_2_bound);
 
-            if    (abs(x) >= taylor_n_bound)
+            if    (std::abs(x) >= taylor_n_bound)
             {
                 return(sin(x)/x);
             }
@@ -76,14 +76,14 @@
                 // approximation by taylor series in x at 0 up to order 0
                 T    result = static_cast<T>(1);
 
-                if    (abs(x) >= taylor_0_bound)
+                if    (std::abs(x) >= taylor_0_bound)
                 {
                     T    x2 = x*x;
 
                     // approximation by taylor series in x at 0 up to order 2
                     result -= x2/static_cast<T>(6);
 
-                    if    (abs(x) >= taylor_2_bound)
+                    if    (std::abs(x) >= taylor_2_bound)
                     {
                         // approximation by taylor series in x at 0 up to order 4
                         result += (x2*x2)/static_cast<T>(120);
diff -ruN boost-1_50_0/tools/build/v2/tools/darwin.jam boost-1_50_0-new/tools/build/v2/tools/darwin.jam
--- boost-1_50_0/tools/build/v2/tools/darwin.jam	2012-04-25 20:35:55.000000000 -0700
+++ boost-1_50_0-new/tools/build/v2/tools/darwin.jam	2012-11-24 01:41:44.000000000 -0800
@@ -183,7 +183,7 @@
     #   archives in darwin is complicated.
     local archiver =
         [ common.get-invocation-command darwin
-            : libtool : [ feature.get-values <archiver> : $(options) ] : $(bin) : search-path ] ;
+            : /usr/bin/libtool : [ feature.get-values <archiver> : $(options) ] : $(bin) : search-path ] ;
     flags darwin.archive .LIBTOOL $(condition) : $(archiver[1]) ;
     if $(.debug-configuration)
     {
