diff -r d3dbef5cbc90 include/Inventor/SbBasic.h
--- a/include/Inventor/SbBasic.h	Fri Feb 28 14:56:02 2014 +0000
+++ b/include/Inventor/SbBasic.h	Thu Mar 27 05:27:51 2014 -0600
@@ -34,6 +34,7 @@
 \**************************************************************************/
 
 #include <Inventor/C/basic.h>
+#include <Inventor/C/errors/debugerror.h>
 
 /* ********************************************************************** */
 /* Trap people trying to use Inventor headers while compiling C source code.
