diff -ruN ./CMakeLists.txt ../gtest-1.5.0-new/CMakeLists.txt
--- ./CMakeLists.txt	2010-04-15 16:02:08.000000000 -0600
+++ ../gtest-1.5.0-new/CMakeLists.txt	2011-01-15 07:50:51.000000000 -0700
@@ -382,3 +382,16 @@
   cxx_executable(gtest_xml_output_unittest_ test gtest)
   py_test(gtest_xml_output_unittest)
 endif()
+
+# JRC: add installation target
+
+INSTALL(TARGETS gtest gtest_main
+  RUNTIME DESTINATION bin
+  LIBRARY DESTINATION lib/${CXX_COMP_LIB_SUBDIR}
+  ARCHIVE DESTINATION lib/${CXX_COMP_LIB_SUBDIR}
+)
+
+INSTALL(FILES
+  ${CMAKE_CURRENT_BINARY_DIR}/CMakeCache.txt
+  DESTINATION share
+)
