diff -ruN ./src/H5public.h ../hdf5-1.8.5-new/src/H5public.h
--- ./src/H5public.h	2010-06-04 12:25:52.000000000 -0600
+++ ../hdf5-1.8.5-new/src/H5public.h	2010-08-07 16:30:41.277100800 -0600
@@ -42,11 +42,9 @@
 #ifdef H5_STDC_HEADERS
 #   include <limits.h>		/*for H5T_NATIVE_CHAR defn in H5Tpublic.h    */
 #endif
-#ifndef __cplusplus
 #ifdef H5_HAVE_STDINT_H
 #   include <stdint.h>		/*for C9x types				     */
 #endif
-#endif
 #ifdef H5_HAVE_INTTYPES_H
 #   include <inttypes.h>        /* For uint64_t on some platforms            */
 #endif
