diff -ruN netcdf_fortran-4.2/configure netcdf_fortran-4.2-new/configure
--- netcdf_fortran-4.2/configure	2011-10-03 08:10:43.000000000 -0600
+++ netcdf_fortran-4.2-new/configure	2015-04-12 14:09:03.000000000 -0600
@@ -17356,7 +17356,7 @@
 $as_echo "#define pgiFortran 1" >>confdefs.h
 
             ;;
-          *gfortran)
+          *gfortran*)
              if (gfortran --version | grep '(GCC 4.1.' || gfortran --version | grep '(GCC 4.0.'); then
 
 $as_echo "#define gFortran 1" >>confdefs.h
diff -ruN netcdf_fortran-4.2/configure.ac netcdf_fortran-4.2-new/configure.ac
--- netcdf_fortran-4.2/configure.ac	2011-10-03 08:10:14.000000000 -0600
+++ netcdf_fortran-4.2-new/configure.ac	2015-04-12 14:09:10.000000000 -0600
@@ -231,7 +231,7 @@
          pgf95|pgf90|pgf77|ifort*|lf90|lf95)
             AC_DEFINE(pgiFortran, [1], [Turned on by netCDF configure.])
             ;;
-          *gfortran) 
+          *gfortran*) 
              if (gfortran --version | grep '(GCC 4.1.' || gfortran --version | grep '(GCC 4.0.'); then
                 AC_DEFINE(gFortran, [1], [Turned on by netCDF configure.])    
              else
