diff -ruN numpy-1.6.1/site.cfg.example numpy-1.6.1-new/site.cfg.example
--- numpy-1.6.1/site.cfg.example	2011-03-10 21:56:16.000000000 -0800
+++ numpy-1.6.1-new/site.cfg.example	2012-06-23 15:56:18.000000000 -0700
@@ -59,9 +59,9 @@
 # This is a good place to add general library and include directories like
 # /usr/local/{lib,include}
 #
-#[DEFAULT]
-#library_dirs = /usr/local/lib
-#include_dirs = /usr/local/include
+[DEFAULT]
+library_dirs = /usr/local/lib
+include_dirs = /usr/local/include
 
 # Optimized BLAS and LAPACK
 # -------------------------
@@ -90,6 +90,12 @@
 #[lapack_opt]
 #libraries = lapack, ptf77blas, ptcblas, atlas
 
+# JRC, 20120623: Regardless of what is said above, the below is what works.
+[blas]
+blas_libs = blas
+
+[lapack]
+lapack_libs = lapack
 
 # UMFPACK
 # -------
